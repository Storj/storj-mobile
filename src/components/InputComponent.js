import {
    View,
    TextInput,
    Text,
    Image,
    StyleSheet,
    TouchableOpacity
} from 'react-native';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getWidth, getHeight } from '../utils/adaptive';

/**
* Styled input component
*/
export default class InputComponent extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isTextShown: this.props.isPassword
        };
    };

    onPress() {
        this.setState({isTextShown: !this.state.isTextShown})
    }

    render() {
        const style = typeof(this.props.style) === "object" ? this.props.style : null; 
        const errorMessage = this.props.errorMessage ? this.props.errorMessage : null; 

        return (
            <View style = { [ styles.inputWrapper, style ] }>
                <View style = { styles.textInputWrapper }>
                    <TextInput 
                        placeholderTextColor = 'grey'
                        placeholder = { this.props.placeholder }
                        editable = { this.props.editable}
                        secureTextEntry = { this.state.isTextShown }
                        style = { [styles.textInput, this.props.inputStyle] } 
                        multiline = { this.props.multiline }
                        underlineColorAndroid = 'transparent'
                        autoCorrect={ false }
                        onChangeText={ this.props.onChangeText }
                        value={ this.props.value } />
                        {
                            (() => {
                                if(this.props.isPassword) {
                                    return (
                                         <TouchableOpacity onPress = { () => { this.onPress() } }>
                                            <Image                     
                                                style = { styles.eye  } 
                                                source = { require('../images/Icons/Eye.png') } 
                                                resizeMode = 'contain'/>
                                        </TouchableOpacity>
                                    );
                                }
                            })()
                        }
                </View>
                <View style = { styles.underline }/>
                <Text>{ this.props.isError ? errorMessage : null }</Text>
            </View>
        );
    };
}

InputComponent.propTypes = {
    onChangeText: PropTypes.func,
    isPassword: PropTypes.bool,
    editable: PropTypes.bool,
    placeholder: PropTypes.string,
    style: PropTypes.object,
    value: PropTypes.string
};

const styles = StyleSheet.create({
    inputWrapper: {
        paddingTop: getHeight(12),
        height: getHeight(74)
    },
    textInputWrapper: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    textInput: {
        height: getHeight(24),
        width: getWidth(281),
        lineHeight: getHeight(24),
        fontSize: getHeight(16),
        color: '#1c1b1b',
        fontFamily: 'Montserrat',
        padding: 0,
        marginBottom: getHeight(8)
    },
    underline: {
        height: getHeight(1),
        backgroundColor: '#a6b5ca'
    },
    eye: {
        width: getHeight(24),
        height: getHeight(24)
    }
});