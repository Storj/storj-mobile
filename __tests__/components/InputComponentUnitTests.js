import 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import InputComponent from '../../src/components/InputComponent';

Enzyme.configure({ adapter: new Adapter() });


describe('InputComponent', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <InputComponent />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with multiline', () => {
        const wrapper = shallow(
            <InputComponent 
                multiline = { true }/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('triggers onChangeText correctly', () => {
        const onChangeTextSpy = jest.fn();

        const wrapper = shallow(
            <InputComponent 
                onChangeText = { onChangeTextSpy } />
        );

        wrapper.find('TextInput').forEach(child => {
            child.simulate('ChangeText');
        });

        expect(onChangeTextSpy.mock.calls.length).toBe(1);
    });

    it('triggers setState on Press', () => {
        const wrapper = shallow(
            <InputComponent 
                isPassword = { true } />
        );

        const handle = wrapper.instance();
        const setStateSpy = sinon.spy(handle, 'setState');

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(setStateSpy.calledOnce).toBe(true);
    });

    it('changes state correctly on Press', () => {
        const wrapper = shallow(
            <InputComponent 
                isPassword = { true } />
        );

        const handle = wrapper.instance();

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(handle.state.isTextShown).toBe(false);
    });
});