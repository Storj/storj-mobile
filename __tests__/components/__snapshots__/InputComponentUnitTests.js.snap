// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#4b657d",
              "fontFamily": "Montserrat",
              "fontSize": 24,
              "fontWeight": "bold",
              "height": 30,
              "opacity": 0.4,
            }
          }
        >
           
           
        </Text>,
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={undefined}
            multiline={undefined}
            onChangeText={undefined}
            placeholder={undefined}
            placeholderTextColor="grey"
            secureTextEntry={undefined}
            style={
              Array [
                Object {
                  "color": "#1c1b1b",
                  "fontFamily": "Montserrat",
                  "fontSize": 32,
                  "height": 48,
                  "lineHeight": 48,
                  "marginBottom": 16,
                  "padding": 0,
                  "width": 562,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value={undefined}
          />
        </View>,
        <View
          style={
            Object {
              "backgroundColor": "#a6b5ca",
              "height": 1,
              "opacity": 0.2,
            }
          }
        />,
      ],
      "style": Array [
        Object {
          "height": 148,
          "paddingTop": 24,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            " ",
            undefined,
            " ",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#4b657d",
            "fontFamily": "Montserrat",
            "fontSize": 24,
            "fontWeight": "bold",
            "height": 30,
            "opacity": 0.4,
          },
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              editable={undefined}
              multiline={undefined}
              onChangeText={undefined}
              placeholder={undefined}
              placeholderTextColor="grey"
              secureTextEntry={undefined}
              style={
                Array [
                  Object {
                    "color": "#1c1b1b",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                    "height": 48,
                    "lineHeight": 48,
                    "marginBottom": 16,
                    "padding": 0,
                    "width": 562,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={undefined}
            />,
            undefined,
          ],
          "style": Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCorrect": false,
              "editable": undefined,
              "multiline": undefined,
              "onChangeText": undefined,
              "placeholder": undefined,
              "placeholderTextColor": "grey",
              "secureTextEntry": undefined,
              "style": Array [
                Object {
                  "color": "#1c1b1b",
                  "fontFamily": "Montserrat",
                  "fontSize": 32,
                  "height": 48,
                  "lineHeight": 48,
                  "marginBottom": 16,
                  "padding": 0,
                  "width": 562,
                },
                undefined,
              ],
              "underlineColorAndroid": "transparent",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "backgroundColor": "#a6b5ca",
            "height": 1,
            "opacity": 0.2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4b657d",
                "fontFamily": "Montserrat",
                "fontSize": 24,
                "fontWeight": "bold",
                "height": 30,
                "opacity": 0.4,
              }
            }
          >
             
             
          </Text>,
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              editable={undefined}
              multiline={undefined}
              onChangeText={undefined}
              placeholder={undefined}
              placeholderTextColor="grey"
              secureTextEntry={undefined}
              style={
                Array [
                  Object {
                    "color": "#1c1b1b",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                    "height": 48,
                    "lineHeight": 48,
                    "marginBottom": 16,
                    "padding": 0,
                    "width": 562,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={undefined}
            />
          </View>,
          <View
            style={
              Object {
                "backgroundColor": "#a6b5ca",
                "height": 1,
                "opacity": 0.2,
              }
            }
          />,
        ],
        "style": Array [
          Object {
            "height": 148,
            "paddingTop": 24,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              " ",
              undefined,
              " ",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#4b657d",
              "fontFamily": "Montserrat",
              "fontSize": 24,
              "fontWeight": "bold",
              "height": 30,
              "opacity": 0.4,
            },
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                editable={undefined}
                multiline={undefined}
                onChangeText={undefined}
                placeholder={undefined}
                placeholderTextColor="grey"
                secureTextEntry={undefined}
                style={
                  Array [
                    Object {
                      "color": "#1c1b1b",
                      "fontFamily": "Montserrat",
                      "fontSize": 32,
                      "height": 48,
                      "lineHeight": 48,
                      "marginBottom": 16,
                      "padding": 0,
                      "width": 562,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value={undefined}
              />,
              undefined,
            ],
            "style": Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": true,
                "autoCorrect": false,
                "editable": undefined,
                "multiline": undefined,
                "onChangeText": undefined,
                "placeholder": undefined,
                "placeholderTextColor": "grey",
                "secureTextEntry": undefined,
                "style": Array [
                  Object {
                    "color": "#1c1b1b",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                    "height": 48,
                    "lineHeight": 48,
                    "marginBottom": 16,
                    "padding": 0,
                    "width": 562,
                  },
                  undefined,
                ],
                "underlineColorAndroid": "transparent",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "backgroundColor": "#a6b5ca",
              "height": 1,
              "opacity": 0.2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InputComponent renders correctly with multiline 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputComponent
    multiline={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#4b657d",
              "fontFamily": "Montserrat",
              "fontSize": 24,
              "fontWeight": "bold",
              "height": 30,
              "opacity": 0.4,
            }
          }
        >
           
           
        </Text>,
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={undefined}
            multiline={true}
            onChangeText={undefined}
            placeholder={undefined}
            placeholderTextColor="grey"
            secureTextEntry={undefined}
            style={
              Array [
                Object {
                  "color": "#1c1b1b",
                  "fontFamily": "Montserrat",
                  "fontSize": 32,
                  "height": 48,
                  "lineHeight": 48,
                  "marginBottom": 16,
                  "padding": 0,
                  "width": 562,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value={undefined}
          />
        </View>,
        <View
          style={
            Object {
              "backgroundColor": "#a6b5ca",
              "height": 1,
              "opacity": 0.2,
            }
          }
        />,
      ],
      "style": Array [
        Object {
          "height": 148,
          "paddingTop": 24,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            " ",
            undefined,
            " ",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#4b657d",
            "fontFamily": "Montserrat",
            "fontSize": 24,
            "fontWeight": "bold",
            "height": 30,
            "opacity": 0.4,
          },
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              editable={undefined}
              multiline={true}
              onChangeText={undefined}
              placeholder={undefined}
              placeholderTextColor="grey"
              secureTextEntry={undefined}
              style={
                Array [
                  Object {
                    "color": "#1c1b1b",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                    "height": 48,
                    "lineHeight": 48,
                    "marginBottom": 16,
                    "padding": 0,
                    "width": 562,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={undefined}
            />,
            undefined,
          ],
          "style": Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCorrect": false,
              "editable": undefined,
              "multiline": true,
              "onChangeText": undefined,
              "placeholder": undefined,
              "placeholderTextColor": "grey",
              "secureTextEntry": undefined,
              "style": Array [
                Object {
                  "color": "#1c1b1b",
                  "fontFamily": "Montserrat",
                  "fontSize": 32,
                  "height": 48,
                  "lineHeight": 48,
                  "marginBottom": 16,
                  "padding": 0,
                  "width": 562,
                },
                undefined,
              ],
              "underlineColorAndroid": "transparent",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "backgroundColor": "#a6b5ca",
            "height": 1,
            "opacity": 0.2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4b657d",
                "fontFamily": "Montserrat",
                "fontSize": 24,
                "fontWeight": "bold",
                "height": 30,
                "opacity": 0.4,
              }
            }
          >
             
             
          </Text>,
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              editable={undefined}
              multiline={true}
              onChangeText={undefined}
              placeholder={undefined}
              placeholderTextColor="grey"
              secureTextEntry={undefined}
              style={
                Array [
                  Object {
                    "color": "#1c1b1b",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                    "height": 48,
                    "lineHeight": 48,
                    "marginBottom": 16,
                    "padding": 0,
                    "width": 562,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value={undefined}
            />
          </View>,
          <View
            style={
              Object {
                "backgroundColor": "#a6b5ca",
                "height": 1,
                "opacity": 0.2,
              }
            }
          />,
        ],
        "style": Array [
          Object {
            "height": 148,
            "paddingTop": 24,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              " ",
              undefined,
              " ",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#4b657d",
              "fontFamily": "Montserrat",
              "fontSize": 24,
              "fontWeight": "bold",
              "height": 30,
              "opacity": 0.4,
            },
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                editable={undefined}
                multiline={true}
                onChangeText={undefined}
                placeholder={undefined}
                placeholderTextColor="grey"
                secureTextEntry={undefined}
                style={
                  Array [
                    Object {
                      "color": "#1c1b1b",
                      "fontFamily": "Montserrat",
                      "fontSize": 32,
                      "height": 48,
                      "lineHeight": 48,
                      "marginBottom": 16,
                      "padding": 0,
                      "width": 562,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value={undefined}
              />,
              undefined,
            ],
            "style": Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": true,
                "autoCorrect": false,
                "editable": undefined,
                "multiline": true,
                "onChangeText": undefined,
                "placeholder": undefined,
                "placeholderTextColor": "grey",
                "secureTextEntry": undefined,
                "style": Array [
                  Object {
                    "color": "#1c1b1b",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                    "height": 48,
                    "lineHeight": 48,
                    "marginBottom": 16,
                    "padding": 0,
                    "width": 562,
                  },
                  undefined,
                ],
                "underlineColorAndroid": "transparent",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "backgroundColor": "#a6b5ca",
              "height": 1,
              "opacity": 0.2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
